src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs

```diff
@@ -159,15 +165,34 @@ public async Task<string> Download(Uri url, bool logPermissionDenied = false)
                 return await response.Content.ReadAsStringAsync();
             }
 
-            if (logPermissionDenied && response.StatusCode == HttpStatusCode.Forbidden)
+            CheckPermissions(response, logPermissionDenied);
+            return null;
+        }
+
+        public async Task<Stream> DownloadStream(Uri url)
+        {
+            logger.LogDebug(Resources.MSG_Downloading, url);
+            var response = await client.GetAsync(url);
+            if (response.IsSuccessStatusCode)
             {
-                logger.LogWarning(Resources.MSG_Forbidden_BrowsePermission);
-                response.EnsureSuccessStatusCode();
+                return await response.Content.ReadAsStreamAsync();
             }
 
+            CheckPermissions(response, true);
```

Comment:
Why `true`? The `Download` defaults to `false`. And that means "do nothing".

The whole logic is rather weird. I would revert it to original state. And here just return `null`